internal/replay/replay_http_server.go

```diff
@@ -122,14 +123,57 @@ func (r *ReplayHTTPServer) createRedactedRequest(req *http.Request) (*store.Reco
 	return recordedRequest, nil
 }
 
-func (r *ReplayHTTPServer) loadResponse(fileName string) (*store.RecordedResponse, error) {
-	responseFile := filepath.Join(r.recordingDir, fileName+".resp")
-	fmt.Printf("loading response from : %s\n", responseFile)
-	responseData, err := os.ReadFile(responseFile)
+func (r *ReplayHTTPServer) loadResponse(fileName string, shaSum string) (*store.RecordedResponse, error) {
+	// 1. Open the replay log file for reading.
+	filePath := filepath.Join(r.recordingDir, fileName+".http.log")
+	fmt.Printf("loading response from : %s with shaSum: %s\n", filePath, shaSum)
+	file, err := os.Open(filePath)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not open file %s: %w", filePath, err)
+	}
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+	expectedKey := shaSum + ".resp"
+	// 2. Scan the file line by line using the reader directly.
+	for {
+		// Read one line, including the newline character.
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				return nil, fmt.Errorf("response with shaSum %s not found in file", shaSum)
+			}
+			return nil, fmt.Errorf("error while reading file: %w", err)
+		}
+		trimmedLine := strings.TrimSpace(line)
+		parts := strings.Fields(trimmedLine)
+		if len(parts) != 2 {
+			continue
+		}
+
+		fileKey := parts[0]
+		sizeStr := parts[1]
+
+		// 3. Check if the key on this line matches our target shaSum.
+		if fileKey == expectedKey {
```

Comment:
What happens when we're trying to reply the second response from a file?  do we need to skip size bytes anyway here?

Reply:
Followed up f2f, I still think swe should skip the entire message if it's not a match.

Reply:
updated to skip not matched message entiredly